extensions/script/CMakeLists.txt

```diff
@@ -68,13 +58,10 @@ if (ENABLE_LUA_SCRIPTING)
     add_definitions(-DLUA_SUPPORT)
 
     file(GLOB LUA_SOURCES  "lua/*.cpp")
-    add_library(minifi-lua-extensions STATIC ${LUA_SOURCES})
-
-    target_link_libraries(minifi-lua-extensions ${LIBMINIFI})
-    target_link_libraries(minifi-lua-extensions ${LUA_LIBRARIES} sol)
-    target_link_libraries(minifi-script-extensions minifi-lua-extensions)
+    target_sources(minifi-script-extensions PRIVATE ${LUA_SOURCES})
 
-    target_compile_features(minifi-lua-extensions PUBLIC cxx_std_14)
+    target_link_libraries(minifi-script-extensions ${LUA_LIBRARIES} sol)
+    target_compile_features(minifi-script-extensions PUBLIC cxx_std_14)
```

Comment:
These used to be in a separate `minifi-lua-extensions` library, now they are part of `minifi-script-extensions`.  Why?

Reply:
tried separating them into `script-core.a` (from `extensions/script/*.cpp`), `lua.a` and `python.a`, and merging them into a dynamic lib, but did not find the right combination of cmake commands to avoid linker errors, the separation would be artificial at best anyway IMO, since e.g. `ExecuteScript.cpp` has a number of if-defs checking for lua and python supports

Reply:
I can imagine scenarios where a user wants lua scripting, but doesn't want to install a python interpreter on their embedded system. Or the other way around. So I don't agree with the separation being "artificial" in the sense of being unnecessary.

I can accept a new limitation because of this feature, if it's not worth your time to fix this issue. In this case, please adapt the main CMakeLists.txt and the bootstrap script to only have one option that toggles scripting support with both python and lua.

Reply:
we don't have to have a python interpreter if we enable only the lua extension, those sources are not compiled

Reply:
Ok, and I just realized that even linking is conditional. Nevermind then.